### PR TITLE
[CI: website] Fix overwriting of original files and add version awareness

### PR DESCRIPTION
**Pull request by grief8**
1. Previous CI does not generate a clean `asterinas/book`, so this PR is proposed to force overwriting of original files.
2. Add automatically updates when we bump Asterinas version numbers .

### CODE REVIEW

File: .github/workflows/update-website.yml [hunk 1]
```diff
@@ -36,6 +37,8 @@ jobs:
           GIT_SSH_COMMAND="ssh -i ./book_publish_key -o UserKnownHostsFile=./known_hosts" git clone git@github.com:asterinas/book.git temp_book
           cd temp_book
           git checkout --orphan newBranch
+          git rm -rf .
```
Comment (tatetian) [inline .github/workflows/update-website.yml:40]: Usually, `git rm -rf <target_dir>` removes the `target_dir`. So taking at its face value, the command `git rm -rf .` looks like it deleting the `.` directory, which happens to be the project root. But obviously, deleting the project root is not what we want. What we want is to delete everything under the project root.

So to improve clarity, I suggest using the following command instead.

```
git rm -rf *
```

It should be equivalent to the original one, although I didn't test it.

**Comment by tatetian** [comment]
@grief8 I suspect the current "Update Website" has a bug that is not addressed by this PR. [The latest run of "Update Website" action](https://github.com/asterinas/asterinas/actions/runs/8825761795/job/24230839931) updates [the book repo](https://github.com/asterinas/book/), but the content does not reflect the latest version of the book in the main repo. Could you dig into the issue?

**Comment by grief8** [comment]
> @grief8 I suspect the current "Update Website" has a bug that is not addressed by this PR. [The latest run of "Update Website" action](https://github.com/asterinas/asterinas/actions/runs/8825761795/job/24230839931) updates [the book repo](https://github.com/asterinas/book/), but the content does not reflect the latest version of the book in the main repo. Could you dig into the issue?

I also find this problem, so this PR adds `cp -r ../asterinas/docs/book/* ./` to move the compiled book to the directory. After the test, now `https://grief8.github.io/book/` is the newest version.

File: .github/workflows/update-website.yml (hunk 1 above)
Comment (grief8) [reply .github/workflows/update-website.yml:40]: It is equivalent in this CI.  And these two commands have some slight differences that `git rm -rf *` would keep the files and directories start with `.` while `git rm -rf .` only keeps `.git`. 

There are no `.` files under `book` and `mdbook` won't generate them, so I will replace it with `git rm -rf *`.

**Comment by tatetian** [comment]
> > @grief8 I suspect the current "Update Website" has a bug that is not addressed by this PR. [The latest run of "Update Website" action](https://github.com/asterinas/asterinas/actions/runs/8825761795/job/24230839931) updates [the book repo](https://github.com/asterinas/book/), but the content does not reflect the latest version of the book in the main repo. Could you dig into the issue?
> 
> I also find this problem, so this PR adds `cp -r ../asterinas/docs/book/* ./` to move the compiled book to the directory. After the test, now `https://grief8.github.io/book/` is the newest version.

Ok. You can update the PR.

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**